plugins/demoplugin.c

```diff
@@ -35,29 +35,19 @@
 #include "geanyplugin.h"	/* plugin API, always comes first */
 #include "Scintilla.h"	/* for the SCNotification struct */
 
-
-/* These items are set by Geany before plugin_init() is called. */
-GeanyPlugin		*geany_plugin;
-GeanyData		*geany_data;
-
-
-/* Check that the running Geany supports the plugin API version used below, and check
- * for binary compatibility. */
-PLUGIN_VERSION_CHECK(147)
-
-/* All plugins must set name, description, version and author. */
-PLUGIN_SET_INFO(_("Demo"), _("Example plugin."), "0.1" , _("The Geany developer team"))
-
-
 static GtkWidget *main_menu_item = NULL;
 /* text to be shown in the plugin dialog */
 static gchar *welcome_text = NULL;
 
 
-
 static gboolean on_editor_notify(GObject *object, GeanyEditor *editor,
 								 SCNotification *nt, gpointer data)
 {
+	/* Geany passes the plugin's GeanyPlugin pointer as user_data if it did not set
+	 * it explicitely. This is specific to PluginCallbacks. */
```

Comment:
yes, but OTOH it allows to easily set the data for all callbacks, otherwise to get anything else than the GeanyPlugin pointer one has to set each callback's data manually.  If it was the data set with `set_data()`, one could change the default easily.
Moreover, all non-trivial plugins not using globals will require some custom structure anyway, which is likely to contain a pointer to the GeanyPlugin (or else it's trivial to add it).